src/jsoncanvas.rs

```diff
@@ -2,34 +2,46 @@ use std::collections::HashMap;
 use std::fmt::{Display, Formatter};
 use std::str::FromStr;
 
+use crate::edge::Edge;
 use crate::node::GenericNodeInfo;
 use crate::node::Node;
-use crate::edge::Edge;
-use crate::NodeId;
 use crate::EdgeId;
+use crate::NodeId;
 
-use serde::{Serialize, Serializer, Deserialize, Deserializer};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde_json;
+use thiserror::Error;
```

Comment:
I didn't know thiserror crate, thanks!

Reply:
It takes a lot of pain out of error handling, and is quite common across the ecosystem. `snafu` is the one but it's much more complicated and I don't really understand it :sweat_smile: You'll also see `anyhow` used in binary crates: thiserror/snafu are for libraries where you need to transmit the error info for someone else to deal with, anyhow is for the outer part of binaries where it doesn't matter what the error is because you're just going to stop execution with a helpful message.